platform/ios/src/MGLAnnotationView.h

```diff
@@ -89,7 +89,7 @@ typedef NS_ENUM(NSUInteger, MGLAnnotationViewDragState) {
  contains a non-`nil` value while the annotation view is visible on the map. If
  the view is queued, waiting to be reused, the value is `nil`.
  */
-@property (nonatomic, readonly, nullable) id <MGLAnnotation> annotation;
+@property (nonatomic, nullable) id <MGLAnnotation> annotation;
```

Comment:
If `setAnnotation` exists publicly, then they can override it in a subclass and have the annotation model change the behavior of the view.